inc/functions/files.php

```diff
@@ -839,12 +853,14 @@ function rocket_clean_domain( $lang = '' ) {
 		*/
 		do_action( 'before_rocket_clean_domain', $root, $lang, $url ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
-		// Delete cache domain files.
-		$dirs = glob( $root . '*', GLOB_NOSORT );
-		if ( $dirs ) {
-			foreach ( $dirs as $dir ) {
-				rocket_rrmdir( $dir, get_rocket_i18n_to_preserve( $lang ) );
+		try {
+			$files = new RegexIterator( $iterator, "/{$file['host']}*{$file['path']}/i" );
+			foreach ( $files as $file ) {
+				rocket_rrmdir( $file, get_rocket_i18n_to_preserve( $lang ) );
```

Comment:
@Tabrisrp Will `get_rocket_i18n_to_preserve( $lang )`return the same value each time? If yes, move it before the `foreach` to be more performant.

Reply:
Also, why does the `foreach` need to be within the `try`?

Reply:
You're right, I cherry-picked the changes from PR #2510 to add them here

Reply:
I think the best thing for this part is merge back develop in this branch once 2510 is done and merged

Reply:
@Tabrisrp I merged `develop` with PR 2510 in this commit https://github.com/wp-media/wp-rocket/pull/2458/commits/71bb6dad18370618dd6f2fb57c9225f578fcccac